### PR TITLE
User model write methods

### PR DESCRIPTION
**Pull request by eliascodes**
#76 

### Changes
* Add first version of write methods to the user model object
* Add tests
* Add `factory-girl` for creating test data.
* Tidy up/simplify typing a little

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -1,3 +1,10 @@
+import { Dictionary, CurriedFunction2, assoc, curry } from 'ramda';
+
+
+type RenameKeys<T extends Dictionary<any>, L extends Dictionary<string>> =
```
Comment (astroash) [inline src/utils/index.ts]: 👌

File: database/seeds/testing/14_test_user.seed.js [hunk 2]
```diff
@@ -16,5 +15,36 @@ exports.seed = (knex) =>
         gender_id: knex('gender').select('gender_id').where({ gender_name: 'female' }),
         disability_id: knex('disability').select('disability_id').where({ disability_name: 'no' }),
         ethnicity_id: knex('ethnicity').select('ethnicity_id').where({ ethnicity_name: 'prefer not to say' }),
+      },
+      {
+        user_name: 'Gordon',
```
[the remark this answers is not in the document]
Comment (eliascodes) [reply database/seeds/testing/14_test_user.seed.js]: Lol, I was continuing the Valve theme. We got Chell repping Aperture, we need someone from Black Mesa. Gordon Freeman is the OG. We can make Alyx next if you like.

File: src/models/user.ts [hunk 3]
```diff
@@ -35,6 +36,46 @@ const ColumnToModel: Map<keyof UserRow, keyof User> = {
...
+const transformForeignKeysToSubQueries = (client: Knex) => evolve({
+  gender_id: (v: string) =>
+    client('gender').select('gender_id').where({ gender_name: v }),
+  disability_id: (v: string) =>
+    client('disability').select('disability_id').where({ disability_name: v }),
+  ethnicity_id: (v: string) =>
+    client('ethnicity').select('ethnicity_id').where({ ethnicity_name: v }),
+});
+
+const dropUnWhereableUserFields = omit([
```
[the remark this answers is not in the document]
Comment (astroash) [reply src/models/user.ts]: @eliascodes can this be changed?

[the remark this answers is not in the document]
Comment (eliascodes) [reply src/models/user.ts]: Yeah, sorry, hadn't got round to addressing this yet.

**Pull request merged**